core/templates/dev/head/services/utilsService.js

```diff
@@ -31,6 +31,15 @@ oppia.factory('utilsService', [function() {
     // http://stackoverflow.com/questions/203739
     isString: function(input) {
       return (typeof input === 'string' || input instanceof String);
+    },
+
+    skipTo: function($event) {
+      var link = $event.target.dataset.link;
+      if (document.getElementById(link)) {
```

Comment:
You may not need some of this after some of the other suggestions, but I just wanted to remark that it's probably cleaner to have "var blah = document.getElementById(link);" and then reference blah thereafter, rather than calling document.getElementById() again and again.